src/Algebra/Graph/Bipartite/AdjacencyMap.hs

```diff
@@ -678,15 +687,144 @@ vertexSet = Set.fromAscList . vertexList
 edgeSet :: (Ord a, Ord b) => AdjacencyMap a b -> Set.Set (a, b)
 edgeSet = Set.fromAscList . edgeList
 
+-- | The sorted /adjacency list/ of the left part of the graph.
+-- Complexity: /O(n + m)/ time and /O(n)/ memory.
+--
+-- @
+-- leftAdjacencyList 'empty'            == []
+-- leftAdjacencyList ('vertices' [] xs) == []
+-- leftAdjacencyList ('vertices' xs []) == [ (x, []) | x <- nub ('sort' xs) ]
+-- leftAdjacencyList ('edge' x y)       == [(x, [y])]
+-- leftAdjacencyList ('star' x ys)      == [(x, 'nub' ('sort' ys))]
+-- @
+leftAdjacencyList :: AdjacencyMap a b -> [(a, [b])]
+leftAdjacencyList (BAM lr _) = [ (v, Set.toAscList us) | (v, us) <- Map.toAscList lr ]
+
+-- | The sorted /adjacency list/ of the right part of the graph.
+-- Complexity: /O(n + m)/ time and /O(n)/ memory.
+--
+-- @
+-- rightAdjacencyList 'empty'            == []
+-- rightAdjacencyList ('vertices' [] xs) == [ (x, []) | x <- 'nub' ('sort' xs) ]
+-- rightAdjacencyList ('vertices' xs []) == []
+-- rightAdjacencyList ('edge' x y)       == [(y, [x])]
+-- rightAdjacencyList ('star' x ys)      == [ (y, [x])  | y <- 'nub' ('sort' ys) ]
+-- @
+rightAdjacencyList :: AdjacencyMap a b -> [(b, [a])]
+rightAdjacencyList (BAM _ rl) = [ (v, Set.toAscList us) | (v, us) <- Map.toAscList rl ]
+
+-- | The sorted /adjacency list/ of the graph.
+-- Complexity: /O(n + m)/ time and /O(n)/ memory.
+--
+-- @
+-- adjacencyList 'empty'            == []
+-- adjacencyList ('vertices' xs ys) == [ (v, []) | v <- (map Left (sort (nub xs))) ++ (map Right (sort (nub ys))) ]
+-- adjacencyList ('edge' x y)       == [(Left x, [Right y]), (Right y, [Left x])]
+-- adjacencyList                  == 'Algebra.Graph.AdjacencyMap.adjacencyList' . 'fromBipartite'
+-- @
+adjacencyList :: AdjacencyMap a b -> [(Either a b, [Either a b])]
+adjacencyList g = [ (Left v, map Right us) | (v, us) <- leftAdjacencyList g ] ++
+                  [ (Right v, map Left us) | (v, us) <- rightAdjacencyList g ]
+
+-- | A list of alternating values of two types. The first type argument denotes
+-- the type of the head.
+--
+-- With @OverloadedLists@ extension it is possible to use standard list
+-- notation to make a 'List' of two coincidential types, e.g.
+-- 
+-- @
+-- [1, 2, 3, 4, 5] :: List Int Int
+-- @
+--
+-- This property is heavily used in the examples below.
+--
+-- The 'Show' instance matches the 'Show' instance for lists.
+--
+-- @
+-- 'show' Empty                              == "[]"
+-- 'show' ([1, 2, 3] :: List Int Int)        == "[1,2,3]"
+-- 'show' (Cons 1 (Cons "a" (Cons 3 Empty))) == "[1,\\"a\\",3]"
+-- @
+data List a b = Empty | Cons a (List b a)
```

Comment:
Perhaps we should follow the convention and use `Nil` instead of `Empty` since we are already using `Cons`. This would also avoid potential confusion with the `empty` graph.